### PR TITLE
feat: Improve Error Handling of classifiers and regressors

### LINKED ISSUE #153: Improve error handling of classifiers and regressors
### Is your feature request related to a problem?

In some cases, error handling is not ideal (see solution for details) in the `safeds.ml` package.

### Desired solution

* Validate inputs of a function as the first step
* Raise an appropriate exception with an appropriate message if issues are found
* Document in the docstring that the exception is raised and under which condition
* Test that the correct exception with the correct message is raised

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

* Example: If non-numerical columns are in a table, an appropriate message should be shown. Users should also be guided towards the `OneHotEncoder` or `LabelEncoder` and we should describe when to use which.

### PR DESCRIPTION
**Pull request by alex-senger**
Closes #153 .

### Summary of Changes

The Error Handling of classifiers and regressors was not ideal. Now every classifier and regressor validates the data, especially when given non-numerical or missing values in the table.

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/7.0.4) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/5222651525)

| Descriptor  |                                Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|-----------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/7.0.4/descriptors/python_black)          |   24|    4|                                                             0|1.52s       |
|❌ PYTHON    |[mypy](https://megalinter.io/7.0.4/descriptors/python_mypy)            |   24|     |[1](https://github.com/Safe-DS/Stdlib/actions/runs/5222651525)|2.17s       |
|✅ PYTHON    |[ruff](https://megalinter.io/7.0.4/descriptors/python_ruff)            |   24|    7|                                                             0|0.04s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/7.0.4/descriptors/repository_git_diff)|yes  |     |no                                                            |0.01s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/5222651525)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

File: src/safeds/exceptions/_data.py [hunk 1]
```diff
@@ -23,8 +23,27 @@ def __init__(self, column_names: list[str]):
 class NonNumericColumnError(Exception):
     """Exception raised for trying to do numerical operations on a non-numerical column."""
 
-    def __init__(self, column_info: str) -> None:
-        super().__init__(f"Tried to do a numerical operation on one or multiple non numerical Columns: \n{column_info}")
+    def __init__(self, column_info: str, help_msg: str | None = None) -> None:
+        line_break = "\n"
```
Comment (jxnior01) [inline src/safeds/exceptions/_data.py:27]: Variable could be left out to improve code readability.

File: src/safeds/exceptions/_data.py [hunk 2]
```diff
@@ -23,8 +23,27 @@ def __init__(self, column_names: list[str]):
...
+        super().__init__(
+            (
+                "Tried to do a numerical operation on one or multiple non-numerical columns:"
+                f" \n{column_info}{line_break + help_msg if help_msg is not None else ''}"
+            ),
```
[the remark this answers is not in the document]
Comment (Marsmaennchen221) [reply src/safeds/exceptions/_data.py:32]: We did it this way so we only print the line break if we have a help message to improve the readability of the error message

File: src/safeds/exceptions/_data.py [hunk 3]
```diff
@@ -23,8 +23,27 @@ def __init__(self, column_names: list[str]):
...
+
+class MissingValuesColumnError(Exception):
+    """Exception raised for trying to do operations on a columns containing missing values."""
+
+    def __init__(self, column_info: str, help_msg: str | None = None) -> None:
+        line_break = "\n"
+        super().__init__(
+            (
+                "Tried to do an operation on one or multiple columns containing missing values:"
+                f" \n{column_info}{line_break + help_msg if help_msg is not None else ''}"
```
[the remark this answers is not in the document]
Comment (Marsmaennchen221) [reply src/safeds/exceptions/_data.py:44]: We did it this way so we only print the line break if we have a help message to improve the readability of the error message

File: src/safeds/exceptions/_data.py (hunk 1 above)
Comment (Marsmaennchen221) [reply src/safeds/exceptions/_data.py:27]: We did it this way so we only print the line break if we have a help message to improve the readability of the error message

**Review by jxnior01 — APPROVED** [review]

**Pull request merged**